auto_invoice_advance/models/sale_order.py

```diff
@@ -9,32 +9,99 @@
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
+    down_payment_id = fields.Many2one('account.move',
+                                      string="Down payment invoice",
+                                      readonly=True,
+                                      help="Auto generated down payment invoice")
+
+    @api.returns('self', lambda value: value.id)
+    def copy(self, default=None):
+        # OVERRIDE
+
+        if default is None:
+            default = {}
+
+        # Do not copy down_payment_id from other sale orders.
+        if self.down_payment_id:
+            default['down_payment_id'] = None
+
+        return super(SaleOrder, self).copy(default)
+
     def _cron_create_advance_invoices(self):
         order_ids = self.search(self._prepare_order_deposit_domain())
```

Comment:
```suggestion
        orders = self.search(self._prepare_order_deposit_domain())
```
It's recordsets, not ids so we should not name it "_ids"